hnn_core/network.py

```diff
@@ -741,28 +606,31 @@ def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
         pos = [self.pos_dict['origin'] for _ in src_gid_ran]
         self._add_cell_type(name, pos)
 
-        # Update connectivity_list
-        nc_dict = {
-            'A_delay': self.delay,
-            'threshold': self.threshold,
-        }
         receptors = ['ampa', 'nmda']
         if drive['type'] == 'gaussian':
             receptors = ['ampa']
+
+        src_range = self.gid_ranges[_long_name(drive['name'])]
         # conn-parameters are for each target cell type
         for target_cell, drive_conn in drive['conn'].items():
+            target_range = self.gid_ranges[_long_name(target_cell)]
             for receptor in receptors:
                 if len(drive_conn[receptor]) > 0:
-                    nc_dict['lamtha'] = drive_conn[
-                        receptor]['lamtha']
-                    nc_dict['A_delay'] = drive_conn[
-                        receptor]['A_delay']
-                    nc_dict['A_weight'] = drive_conn[
-                        receptor]['A_weight']
-                    loc = drive_conn['location']
-                    self._all_to_all_connect(
-                        drive['name'], target_cell, loc, receptor,
-                        deepcopy(nc_dict), unique=drive['cell_specific'])
+                    if drive['cell_specific']:
+                        src_gids = self.gid_ranges[_long_name(drive['name'])]
+                        src_start = src_gids[0]  # Necessary for unique feeds
+                        src_gids = [src_gid + src_start for
+                                    src_gid in target_range]
```

Comment:
is this still needed?

Reply:
Oh for sure. Beforehand, creating this was handled by `_all_to_all_connect`

Reply:
but shouldn't you be able to just do:

```py
src_gids  = drive['name']
```

and then `cell.add_connection` will know the right source gids from `self.gid_ranges([drive['name'])` ? 

I am assuming cell specific drives have `len(self.gid_ranges([drive['name'])) == len(target_gids)` ?

Reply:
Oh you're totally right, let me double check to make sure everything matches up between the two and I'll update is if so.

Reply:
I'm actually really glad you brought this up, the current implementation here actually misassigns gids as their `src_gid` will only come from the beginning of the range of the artifical cells. I'm going to try and add a test to check this

Reply:
yeah, something felt fishy about it!